### PR TITLE
improve volcli errors

### PR DESCRIPTION
**Pull request by mapuri**
This change:
- improves the argument count related error string to be more explicit
- adds two List clis where the argument count check was missing to make it consistent with other places.

/cc @erikh @unclejack

### CODE REVIEW

**Comment by erikh** [comment]
I'm going to try this patch out by hand. I just want to see this in action is all, so give me a day or so to do that.

**Comment by mapuri** [comment]
sgtm, thanks @erikh 

actually may be I can add unit-tests for the error strings let me look into that as well

**Comment by mapuri** [comment]
@erikh I have updated the diff to add a few unit tests for error return paths. I had to refactor the functions a bit to make it easier to unit-test. PTAL and let me know if I didn't go nuts here :)

File: volcli/volcli_test.go [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+)
+
+type volcliSuite struct {
+}
+
+var _ = Suite(&volcliSuite{})
+
+func TestVolcli(t *T) { TestingT(t) }
+
+func (s *volcliSuite) SetUpTest(c *C) {
```
Comment (erikh) [inline volcli/volcli_test.go]: these shouldn't be necessary, you can remove them

**Comment by erikh** [comment]
seeing an issue with `systemtestSuite.TestIntegratedRemoveWhileMount`. you can run:

`godep go test -v ./systemtests -check.v -check.f systemtestSuite.TestIntegratedRemoveWhileMount` to run it independently.

**Comment by erikh** [comment]
I love this! Let's get it merged. First, I need to fix some things in the volplugin test runs though.

**Comment by mapuri** [comment]
thanks @erikh . I have taken care of the review comment. I am trying to debug the system test failure but it seems unrelated to this commit.

**Comment by erikh** [comment]
it is, disregard it, I'm working on it.

**Comment by mapuri** [comment]
cool, thanks @erikh. I have squashed the commits, please feel free to merge as appropriate.

**Comment by erikh** [comment]
the same test is failing, which makes me think there's a bug in this patch. this test calls `volcli` in a number of spots, can you validate that it is not broken by your patch?

**Comment by mapuri** [comment]
@erikh 

I have identified the issue, basically in `volumeRemove()` when Post to volmaster is done and it is going to return the 500 status, that error was not translated to the volcli error. This got exposed now as I replaced `errExit()` calls with `return` instead.

 I have updated the patch with the fix and re-triggered the sanity. PTAL

**Comment by erikh** [comment]
awesome thanks!

**Pull request merged**